cmd/gitserver/server/gitservice.go

```diff
@@ -70,9 +72,20 @@ func (s *Server) gitServiceHandler() *gitservice.Handler {
 				metricServiceDuration.WithLabelValues(svc, errLabel).Observe(time.Since(start).Seconds())
 
 				if err != nil {
-					log15.Error("gitservice.ServeHTTP", "svc", svc, "repo", repo, "protocol", protocol, "duration", time.Since(start), "error", err.Error())
+					s.Logger.Error("gitservice.ServeHTTP",
+						log.String("svc", svc),
+						log.String("repo", repo),
+						log.String("protocol", protocol),
+						log.Duration("duration", time.Since(start)),
```

Comment:
It might be less verbose to extract these into a logger created before this condition:

```suggestion
logger := s.Logger.With(
						log.String("svc", svc),
						log.String("repo", repo),
						log.String("protocol", protocol),
						log.Duration("duration", time.Since(start))
```

Reply:
I think there's a few other similar instances